server/channels/app/authorization.go

```diff
@@ -140,55 +131,34 @@ func (a *App) SessionHasPermissionToChannels(c request.CTX, session model.Sessio
 		}
 	}
 
-	if session.IsUnrestricted() {
+	// if System Roles (ie. Admin, TeamAdmin) allow permissions
+	// if so, no reason to check team
+	if a.SessionHasPermissionTo(session, permission) {
+		// make sure all channels exist, otherwise return false.
+		for _, channelID := range channelIDs {
+			_, appErr := a.GetChannel(c, channelID)
+			if appErr != nil && appErr.StatusCode == http.StatusNotFound {
+				return false
+			}
+		}
```

Comment:
We are verifying the channelIDs here, but not the teamIDs in the other method. Intentional? Also, note that the correct channelID is automatically matched with in L152. So there might not be a need for this?

Reply:
For both of these functions...I tried to make sure I matched the functionality in their equivalent single item function. Since `SessionHasPermissionToTeam` doesn't check the team, I didn't add it to `SessionHasPermissionToTeams`. But because `SessionHasPermissionToChannel` checks it before checking `SessionHasPermissionTo`, I added it here.

In the case of a Session Role (system_admin) granting the permission, the function is returned and L152 isn't hit.